### PR TITLE
Fixing travis-ci/build case and stats.py --build --files

### PR DESCRIPTION
**Pull request by A-Hilaly**
@ryukinix can you review this ?
normal PR passed ill try to PR from this to master using command line but I found that [git documentation ](https://git-scm.com/docs/git-request-pull)says that there is a `git request-pull `I'm a bit confused about what `git push -u` do exactly ( I thought its opening pull request on GitHub ) . Can you do it please ? Just open PR with both builds so we see outputs then we try to add / remove solutions trying to loop most of cases. (This needs to be merged before opening PR from /test-branche)

### CODE REVIEW

**Comment by ryukinix** [comment]
I know that this is confuse, but `git request-pull` is not related to `Git PullRequests`. To use features of github (only github platform) you need another command line software called `hub`. On [hub] I use `hub pull-request`. 

[hub]: https://github.com/github/hub

**Comment by ryukinix** [comment]
> Can you do it please ? Just open PR with both builds so we see outputs then we try to add / remove solutions trying to loop most of cases. (This needs to be merged before opening PR from /test-branche)

Yes, I can do it. Just to you know, this is the pass for this special case when you have the main branch on a external (fork) repo:

```
git remote add hilaly https://github.com/A-Hilaly/ProjectEuler.git
git fetch hilaly
git checkout -b hilaly-master hilaly/master
git push -u origin hilaly-master
```

**Comment by ryukinix** [comment]
As this branch is not related to this pull request, you need to check that event here:  https://github.com/DestructHub/ProjectEuler/commits/hilaly-master

**Comment by ryukinix** [comment]
On PR build we have:
```
55.29s$ docker-compose run diff-sync

Creating network "projecteuler_default" with the default driver

Pulling diff-sync (destructhub/project_euler:latest)...

latest: Pulling from destructhub/project_euler

Digest: sha256:f642c5e9ac8c6777d0d1f70e42a5524ca0834854b25a4c5efe8379f8e5163a32

Status: Downloaded newer image for destructhub/project_euler:latest

Command: Files Build

Changed_core_files : 

 ['stats.py', 'test']
```

By another hand, on Push build:
```
$ docker-compose run diff-sync

Creating network "projecteuler_default" with the default driver

Pulling diff-sync (destructhub/project_euler:latest)...

latest: Pulling from destructhub/project_euler

Digest: sha256:f642c5e9ac8c6777d0d1f70e42a5524ca0834854b25a4c5efe8379f8e5163a32

Status: Downloaded newer image for destructhub/project_euler:latest

fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.

Use '--' to separate paths from revisions, like this:

'git <command> [<revision>...] -- [<file>...]'

Command: Build

   (⣾) t: 0.01

Empty DataFrame

Columns: [Problem, Language, Time, Answer, Correct]

Index: []
```

Which the problem here is: `fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.`
So we don't have that branch on this environment. Would be nice if we can easily plug a ssh terminal on travis build... I think I saw that feature in some place...

**Comment by ryukinix** [comment]
` (This needs to be merged before opening PR from /test-branche)` sorry, I missed that. I'm merging.

File: test [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+if [ ! "$TRAVIS_BRANCH" = "" ]; then
```
Comment (ryukinix) [inline test]: I missed that, but probably `[ -v TRAVIS_BRANCH ]` will be enough.

**Comment by ryukinix** [comment]
It's probably better lock the PullRequests for at least with one revision to avoid myself merging to quickly the PRs... (sorry again for that, I know)

**Comment by ryukinix** [comment]
Hmmm... are you creating a new branch for PR, I only saw now. This PR is merging into `testing-branch` not in the `master`... nice. My mistake is not so serious. I'll re-open this PR now from `testing-branch` to `master` and we can both review that, ok?

**Comment by A-Hilaly** [comment]
> Yes, I can do it. Just to you know, this is the step-by-step for this special case when you have the main branch on a external (fork) repo:

I got it !thank you for clearing up

> Which the problem here is: fatal: ambiguous argument 'origin/master': unknown revision or path not in the working tree.
So we don't have that branch on this environment. Would be nice if we can easily plug a ssh terminal on travis build... I think I saw that feature in some place...

Yes :s can  [ -v TRAVIS_BRANCH ] change the out put ? 
Im wondering if travis really `push` the commits to the master to test them. if this case is true, can we check with `git something are we on master ?`

No worries all fine now we can work on same branch !

**Comment by A-Hilaly** [comment]
I just see what I did with TRAVIS_BRANCH rofl ...

**Pull request merged**